src/main/java/net/zhuoweizhang/raspberryjuice/RemoteSession.java

```diff
@@ -209,6 +224,52 @@ protected void handleCommand(String c, String[] args) {
 					send(e.getName());
 				}
 				
+			// world.getEntities
+			} else if (c.equals("world.getEntities")) {
+				StringBuilder bdr = new StringBuilder();				
+				for (Entity e : world.getEntities()) {
+					if ( e.getType().isSpawnable() && e.getType().getTypeId() >= 0 ) {
+						bdr.append(e.getEntityId());
+						bdr.append(",");
+						bdr.append(e.getType().getTypeId());
+						bdr.append(",");
+						bdr.append(e.getType().toString());
+						bdr.append(",");
+						bdr.append(e.getLocation().getX());
+						bdr.append(",");
+						bdr.append(e.getLocation().getY());
+						bdr.append(",");
+						bdr.append(e.getLocation().getZ());
+						bdr.append("|");
+					}
+				}
+				send(bdr.toString());
+				
+			// world.removeEntity
+			} else if (c.equals("world.removeEntity")) {
+				int result = 0;
+				for (Entity e : world.getEntities()) {
+					if (e.getEntityId() == Integer.parseInt(args[0]))
+					{
+						e.remove();
+						result = 1;
+						break;
+					}
+				}
+				send(result);
+				
+			// world.removeEntityType
+			} else if (c.equals("world.removeEntityType")) {
```

Comment:
Why have a specific command to remove an entities of a specific type? The same could be achieved by getting the entities, looping through and removing the entities?

An objective of this project is to keep it simple.

Reply:
Yes, but this approach simplify removing entities. I'm trying to learn Python kids aged from 7 till 10. I wanted to gave them abilitity to remove entity simple like spawEntity method. Looping over list of entities, removing selected ones is a little bit harder to do for small beginners. Look at small sample below. The goal was: write the code which will remove Creeper before it explode near you:

import mcpi.minecraft as minecraft
import time
mc = minecraft.Minecraft.create("127.0.0.1")
player = "MST"
playerEntityId = mc.getPlayerEntityId(player)
while True:
  mc.entity.removeEntityType(playerEntityId,50,2)     # 50 = creeper, distance = 2 blocks
  time.sleep(0.1)

It's simple as much as possible. BTW. removeEntityType method returns how many entities were removed. This way, my little students were able to implement simple multiplayer game. Every player begin in the middle of specified area. Then I spaw a lot of Creepers around it in random locations. The goal is to catch as many creepers as you can just by running near them. Every child counts removed Creepers in his own code, and when time is out they publish score to chat to be able to compare results.

import mcpi.minecraft as minecraft
import time
mc = minecraft.Minecraft.create("127.0.0.1")
player = "MST"
playerEntityId = mc.getPlayerEntityId(player)
score = 0
timer = 0
timeout = 200
mc.postToChat(player + " started the game")
while timer < timeout:
  removed = mc.entity.removeEntityType(playerEntityId,50,2)
  score = score + removed
  timer = timer + 1
  time.sleep(0.1)
mc.postToChat(player + " finished with score: " + str(score))

Again, simple as possible, but believe me, there is so much fun for about ten kids playing it at the same time in the same room. It does no matter that they can individually cheat by inreasing score, increase distance with removeEntityType method, or just publish fake score at the end of game. Lerning Python in such age must be fun. And of course we had conversation about how can we cheat with this simple game. Believe me it wasn't obvious for them :) That's why I like removeEntityType so much.

Reply:
I think the challenge I have with `removeEntityType` as a method within the API is that it is very specific and as I mentioned the same could be achieved within a helper function in the Python module.

As an alternative, what about a method called `removeEntities` - the opposite to `getEntities` which has optional parameters including type and distance. This would offer additional functionality and the option for expansion.